l5kit/l5kit/evaluation/metrics.py

```diff
@@ -303,3 +305,131 @@ def final_displacement_error_mean(
     """
 
     return _final_displacement_error(ground_truth, pred, confidences, avails, "mean")
+
+# TODO(perone): these functions were moved from the closed-loop evaluator.
+# The functions _ego_agent_within_range, _get_boundingbox, _get_sides would ideally
+# be moved to an abstraction of the Agent later. Such as in the example below:
+#
+#  ego = Agent(...)
+#  agent = Agent(...)
+#  bbox = ego.get_boundingbox()
+#  within_range = ego.within_range(agent)
+#  sides = ego.get_sides()
+
+
+def _ego_agent_within_range(ego_centroid: np.ndarray, ego_extent: np.ndarray,
```

Comment:
the original function was vectorised over agents, this one is not. Is the idea to refactor it later when we design the metric framework?

Reply:
This one is vectorized, right ? The one in `avresearch` is not.

Reply:
ah yes sorry. The one in avresearch is not, however the one I ported is vectorised (but in a different way).

A far as I can see from the signature this function works with N ego and N agents, is that correct? In general we can't assume that each ego will have N agents though (the number is variable)

Reply:
It can work with N ego and M agents, sorry the doc wasn't very clear, I'll fix it and use it on the `detect_collision` and submit another PR, should be clear then.

Reply:
In general, you will have an array of N ego and N list of agents where each one has a different length